From 0d286db0c28beddf165e5b3e3a5d9a1510e1d357 Mon Sep 17 00:00:00 2001
From: Tobias Fella <fella@posteo.de>
Date: Wed, 11 Dec 2024 17:16:10 +0100
Subject: [PATCH] Fix crash when sending messages

ECM recently started adding -fhardened, which makes us crash here since we're doing things that aren't valid, but happened to work out fine previously.
---
 src/chatbarcache.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/chatbarcache.cpp b/src/chatbarcache.cpp
index e87d272944..1161e042c9 100644
--- a/src/chatbarcache.cpp
+++ b/src/chatbarcache.cpp
@@ -319,7 +319,8 @@ void ChatBarCache::postMessage()
         return;
     }
 
-    room->postMessage(text(), sendText, *std::get<std::optional<Quotient::RoomMessageEvent::MsgType>>(result), replyId(), editId(), threadId());
+    auto type = std::get<std::optional<Quotient::RoomMessageEvent::MsgType>>(result);
+    room->postMessage(text(), sendText, type ? *type : Quotient::RoomMessageEvent::MsgType::Text, replyId(), editId(), threadId());
     clearCache();
 }
 
-- 
GitLab

